qt/scientific_interfaces/Indirect/IndirectInterface.cpp

```diff
@@ -13,29 +14,31 @@ using namespace MantidQt::API;
 
 namespace MantidQt::CustomInterfaces {
 
-IndirectInterface::IndirectInterface(QWidget *parent)
-    : UserSubWindow(parent),
-      m_settings(dynamic_cast<IndirectSettings *>(InterfaceManager().createSubWindow("Settings", this))) {
+IndirectInterface::IndirectInterface(QWidget *parent) : UserSubWindow(parent) {}
 
-  connect(m_settings.get(), SIGNAL(applySettings()), this, SLOT(applySettings()));
+void IndirectInterface::initLayout() {
+  // Needed to initially apply the settings loaded on the settings GUI
+  applySettings();
 }
 
 void IndirectInterface::help() {
   HelpWindow::showCustomInterface(QString::fromStdString(documentationPage()), QString("indirect"));
 }
 
 void IndirectInterface::settings() {
-  m_settings->loadSettings();
-  m_settings->setWindowModality(Qt::WindowModal);
-  m_settings->show();
+  auto subWindow = InterfaceManager().createSubWindow("Settings", this);
+  auto settingsWindow = dynamic_cast<IndirectSettings *>(subWindow);
+  settingsWindow->connectInterface(this);
+
+  settingsWindow->loadSettings();
+  settingsWindow->setWindowModality(Qt::WindowModal);
+  settingsWindow->show();
 }
 
-void IndirectInterface::applySettings() { applySettings(getInterfaceSettings()); }
+void IndirectInterface::applySettings() { applySettings(IndirectSettings::getSettings()); }
```

Comment:
I found this line interesting because you added a default constructor with no arguments to call another constructor with the settings argument. Is there a benefit in doing this rather than skipping the step of calling a second constructor with the settings argument? Maybe because it gives the functionality of applying settings that come from elsewhere?

Reply:
I believe I did this just for the convenience of not needing to called `IndirectSettings::getSettings()` in each location I want to apply the settings. Overloading functions is very useful when you have two (or more) functions that do the same thing, but they have different parameter arguments https://www.geeksforgeeks.org/function-overloading-c/